### PR TITLE
Make the URLs used by Google Data Transport available in a Public API.

### PR DESCRIPTION
**Pull request by visumickey**
Make the Google Data Transport URLs available as a Public API + Make Fireperf use the public API to skip instrumenting the GoogleDataTransport URLs.

### CODE REVIEW

**Review by maksymmalyhin — APPROVED** [review]
Great, thank you for making the change! Please make sure CI green before merging.

**Comment by maksymmalyhin** [comment]
Looks like the [tests](https://cs.opensource.google/firebase-sdk/firebase-ios-sdk/+/master:GoogleDataTransport/GDTCCTTests/Unit/GDTCCTUploaderTest.m;l=76?q=GDTCCTUploaderTest&ss=firebase-sdk%2Ffirebase-ios-sdk) should be moved to something like `GDTCOREndpointsTests` to fix CI.

**Review by jeremyjiang-dev — APPROVED** [review]
LGTM! Thank you!

File: FirebasePerformance/Sources/FPRURLFilter.m [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 #import "FirebasePerformance/Sources/FPRConsoleLogger.h"
 
-#import "GoogleDataTransport/GDTCCTLibrary/Private/GDTCCTUploader.h"
+#import "GoogleDataTransport/GDTCORLibrary/Internal/GoogleDataTransportInternal.h"
```
Comment (paulb777) [inline FirebasePerformance/Sources/FPRURLFilter.m:20]: This line is needed in the Perf podspec - https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseCrashlytics.podspec#L29

**Review by paulb777 — COMMENTED** [review]
Update the GDT podspec to 8.2.0 and the Perf GDT dep to 8.2

File: GoogleDataTransport/GDTCORLibrary/GDTCOREndpoints.m [hunk 2]
```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2018 Google
```
Comment (zijianjoy) [inline GoogleDataTransport/GDTCORLibrary/GDTCOREndpoints.m:2]: Should the copyright be 2020?

**Review by zijianjoy — APPROVED** [review]
There is an issue on testing:

`
    - ERROR | [OSX] xcodebuild:  /Users/runner/work/firebase-ios-sdk/firebase-ios-sdk/GoogleDataTransport/GDTCCTTests/Unit/GDTCCTUploaderTest.m:78:60: error: property 'uploadURLs' not found on object of type 'GDTCCTUploader'`

File: GoogleDataTransport/GDTCORLibrary/GDTCOREndpoints.m (hunk 2 above)
Comment (maksymmalyhin) [reply GoogleDataTransport/GDTCORLibrary/GDTCOREndpoints.m:2]: Yeah, it should be `Copyright 2020 Google LLC`, thank you for catching this.

Comment (visumickey) [reply GoogleDataTransport/GDTCORLibrary/GDTCOREndpoints.m]: Done.

File: FirebasePerformance/Sources/FPRURLFilter.m (hunk 1 above)
Comment (visumickey) [reply FirebasePerformance/Sources/FPRURLFilter.m:20]: This is done.

**Comment by visumickey** [comment]
> Update the GDT podspec to 8.2.0 and the Perf GDT dep to 8.2

This is done.

**Comment by visumickey** [comment]
> Looks like the [tests](https://cs.opensource.google/firebase-sdk/firebase-ios-sdk/+/master:GoogleDataTransport/GDTCCTTests/Unit/GDTCCTUploaderTest.m;l=76?q=GDTCCTUploaderTest&ss=firebase-sdk%2Ffirebase-ios-sdk) should be moved to something like `GDTCOREndpointsTests` to fix CI.

Tests are moved.

File: GoogleDataTransport/GDTCCTLibrary/Protogen/nanopb/cct.nanopb.c [hunk 3]
```diff
@@ -17,7 +17,7 @@
 /* Automatically generated nanopb constant definitions */
 /* Generated by nanopb-0.3.9.7 */
 
-#include "GoogleDataTransport/GDTCCTLibrary/Protogen/nanopb/cct.nanopb.h"
+#include "cct.nanopb.h"
```
Comment (maksymmalyhin) [inline GoogleDataTransport/GDTCCTLibrary/Protogen/nanopb/cct.nanopb.c:20]: Please revert this.

File: GoogleDataTransport/GDTCORLibrary/GDTCOREndpoints.m [hunk 4]
```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2020 Google
```
Comment (maksymmalyhin) [inline GoogleDataTransport/GDTCORLibrary/GDTCOREndpoints.m:2]: Please add `LLC` here and at other places.

File: GoogleDataTransport/GDTCORTests/Unit/GDTCOREndpointsTest.m [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  XCTAssertNotNil([GDTCOREndpoints uploadURLForTarget:kGDTCORTargetCCT]);
+  XCTAssertNotNil([GDTCOREndpoints uploadURLForTarget:kGDTCORTargetFLL]);
+  XCTAssertNotNil([GDTCOREndpoints uploadURLForTarget:kGDTCORTargetCSH]);
+  XCTAssertNotNil([GDTCOREndpoints uploadURLForTarget:kGDTCORTargetINT]);
+}
+
+/* Verify if the upload URLs are correct for different endpoints. */
+- (void)testUploadURLsAreCorrect {
+  NSDictionary<NSNumber *, NSURL *> *uploadURLs = [GDTCOREndpoints uploadURLs];
+  XCTAssertEqualObjects([GDTCOREndpoints uploadURLForTarget:kGDTCORTargetCCT],
```
Comment (maksymmalyhin) [inline GoogleDataTransport/GDTCORTests/Unit/GDTCOREndpointsTest.m:39]: nit: it's an interesting test. We are kind of checking the class `GDTCOREndpoints` by its own output. I think we should supplement this test by at least checking that the URLs are not `nil`. The previous version of the test looks a bit more useful to me, though I'm not too strong about it.

**Review by maksymmalyhin — APPROVED** [review]
Mostly LGTM, a couple minor comments.

File: GoogleDataTransport/GDTCORTests/Unit/GDTCOREndpointsTest.m [hunk 6]
```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2018 Google
```
Comment (paulb777) [inline GoogleDataTransport/GDTCORTests/Unit/GDTCOREndpointsTest.m:2]: 2020 Google LLC

**Review by paulb777 — APPROVED** [review]
LGTM on comments resolution and green CI

File: GoogleDataTransport/GDTCORTests/Unit/GDTCOREndpointsTest.m (hunk 5 above)
Comment (visumickey) [reply GoogleDataTransport/GDTCORTests/Unit/GDTCOREndpointsTest.m:39]: There is a separate test to verify that the URLs are not NIL. The previous version is very close to what we have here.

File: GoogleDataTransport/GDTCORTests/Unit/GDTCOREndpointsTest.m (hunk 6 above)
Comment (visumickey) [reply GoogleDataTransport/GDTCORTests/Unit/GDTCOREndpointsTest.m:2]: Done.

File: GoogleDataTransport/GDTCORLibrary/GDTCOREndpoints.m (hunk 4 above)
Comment (visumickey) [reply GoogleDataTransport/GDTCORLibrary/GDTCOREndpoints.m]: Done.

File: GoogleDataTransport/GDTCCTLibrary/Protogen/nanopb/cct.nanopb.c (hunk 3 above)
Comment (visumickey) [reply GoogleDataTransport/GDTCCTLibrary/Protogen/nanopb/cct.nanopb.c]: Done.

**Review by jeremyjiang-dev — APPROVED** [review]

**Pull request merged**